src/Project.hs

```diff
@@ -87,12 +89,24 @@ data PullRequestStatus
   | PrStatusFailedBuild      -- Integrated, but the build failed.
   deriving (Eq)
 
+-- A PR can currently be approved to be merged with "<prefix> merge".
+-- Later on we intend to add different approval commands that trigger different
+-- behaviour, and this enumeration will distinguish these cases.
+data ApprovedFor
+  = Merge
+  -- | MergeAndDeploy
+  deriving (Eq, Show, Generic)
+
+-- For a PR to be approved a specific user must give a specific approval
+-- command, i.e. either "just merge" or "merge and deploy".
+type Approval = (Username, ApprovedFor)
```

Comment:
I would make this an ADT instead of a `type` for extra type safety.

Reply:
So something like:

```haskell
data Approval =
  { approvedBy :: Username
  , approvedFor :: ApprovedFor
  }
```

Then?

Reply:
Yes, exactly.

Reply:
Done. I renamed the `approvedBy` field to `approver` to avoid shadowing as there are some places where `approvedBy` is used as a variable name.